src/app/child-dev-project/children/child-photo-service/datatype-photo.spec.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ *     This file is part of ndb-core.
+ *
+ *     ndb-core is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     ndb-core is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with ndb-core.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { TestBed } from "@angular/core/testing";
+import { SafeUrl } from "@angular/platform-browser";
+import { EntitySchemaService } from "../../../core/entity/schema/entity-schema.service";
+import { DatabaseField } from "../../../core/entity/database-field.decorator";
+import { Entity } from "../../../core/entity/entity";
+import { ChildPhotoService } from "./child-photo.service";
+import { BehaviorSubject } from "rxjs";
+import { PhotoDatatype } from "./datatype-photo";
+import { Photo } from "./photo";
+import { Child } from "../model/child";
+
+describe("dataType photo", () => {
+  let entitySchemaService: EntitySchemaService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [EntitySchemaService],
+    });
+
+    entitySchemaService = TestBed.inject<EntitySchemaService>(
+      EntitySchemaService
+    );
+    entitySchemaService.registerSchemaDatatype(new PhotoDatatype());
+  });
+
+  it("should only save the path of an image", () => {
+    class TestEntity extends Entity {
+      @DatabaseField({ dataType: "photo" }) photo: Photo;
+    }
+    const id = "test1";
+    const entity = new TestEntity(id);
+    entity.photo = { path: "12345", photo: new BehaviorSubject<SafeUrl>(null) };
+
+    const rawData = entitySchemaService.transformEntityToDatabaseFormat(entity);
+    expect(rawData.photo).toEqual(entity.photo.path);
+  });
+
+  it("should set the default photo when loading", () => {
+    class TestEntity extends Entity {
+      @DatabaseField({ dataType: "photo", defaultValue: true })
```

Comment:
the way it's used in `Child` class, I would expect the `defaultValue` to become the value, not just trigger some processing in this case?

Reply:
I changed it now to `defaultValue = ""` in both cases. The default value is really tricky here because no matter which value is set at the beginning `assets/child-photos/` will be appended. This means setting `"default.png"` as default value will end up as `"assets/child-photos/default.png"`. So currently this whole component is still a little weird because it is very dependent on the Child entity. Once we want to actually allow this for other entities as well, this will have to be further adapted.